src/wasm.rs

```diff
@@ -86,6 +86,36 @@ mod fetch {
 
     use crate::Error;
 
+    enum WindowOrWorker {
```

Comment:
would it make sense to call this something like GlobalScope?

Reply:
Thanks for the review, this was adapted from the gloo[2] crate.

I think semantically, it would make sense to only use this as a intermediate caller, since worker and window scope are directly taken from various web api, so they are not really the same.

More over, there are actually 6 kinds of GlobalScope[1] in the web-sys crate all used to interact with different API's, despite that only two of them can call fetch.

[1]: https://docs.rs/web-sys/0.3.45/web_sys/?search=fetch_with_
[2]: https://github.com/rustwasm/gloo/blob/f9b1727481125f1f2e3d052137f92d1e6de9af93/crates/timers/src/callback.rs#L12

Reply:
[additional reference](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope)

Reply:
@jbr Is this good enough to merge? Would like to try it again with surf and cloudflare workers.